runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/WorkerCustomSourcesTest.java

```diff
@@ -197,13 +199,26 @@ private static Work createMockWork(Windmill.WorkItem workItem, Watermarks waterm
         workItem,
         watermarks,
         Work.createProcessingContext(
-            COMPUTATION_ID,
-            (a, b) -> Windmill.KeyedGetDataResponse.getDefaultInstance(),
-            ignored -> {}),
+            COMPUTATION_ID, createMockGetDataClient(), ignored -> {}, mock(HeartbeatSender.class)),
         Instant::now,
         Collections.emptyList());
   }
 
+  private static GetDataClient createMockGetDataClient() {
```

Comment:
why does mock(GetDataClient) not work? 
ditto for other tests

if it doesn't for some reason, can we have a shared factory for this for tests?